tests/test_connector_matrix.py

```diff
@@ -249,6 +253,20 @@ async def test_connect(self):
         ) as patched_get_nick, amock.patch(
             api_string.format("set_displayname")
         ) as patch_set_nick, amock.patch(
+            api_string.format("send_to_device_messages")
+        ) as patched_send_to_device, amock.patch(
+            api_string.format("should_upload_keys")
+        ) as patched_should_upload, amock.patch(
+            api_string.format("should_query_keys")
+        ) as patched_should_query, amock.patch(
+            api_string.format("keys_upload")
+        ) as patched_keys_upload, amock.patch(
+            api_string.format("keys_query")
+        ) as patched_keys_query, amock.patch(
+            "pathlib.Path.mkdir"
+        ) as patched_mkdir, amock.patch(
+            "pathlib.Path.is_dir"
+        ) as patched_is_dir, amock.patch(
             "aiohttp.ClientSession"
         ) as patch_cs, OpsDroid() as _:
```

Comment:
oh my this list of mocks has got out of control :rofl: 

Do you fancy refactoring this to use the mocker fixture? You can make fixtures which use mocker to mock things or groups of things and then pass them to tests, so for this you could mock all the e2ee related things in one fixture like this:

```python
def e2ee_mock(mocker):
    mocker.patch(api_string.format("send_to_device_messages"))
    ....
```
the only thing with this approach is returning the patched object for later assertions, maybe a dict of `{'api_string': patched}` would be the best way to handle that.

Or if that's a little too hacky, you could at least use the mocker fixture in this test to move all of these from being context managers to function calls which would make it read a lot nicer.